tgui/packages/tgui/interfaces/MinimapController.js

```diff
@@ -0,0 +1,245 @@
+    image,
+    x,
+    y,
+  } = data;
+
+  const [name, setName] = useLocalState(context, 'name');
+
+  const toggleNewMarkerMenu = () => {
+    data.showNewMarkerMenu = !showNewMarkerMenu;
+  };
+
+  const newMarker = () => {
+    toggleNewMarkerMenu();
+    act('new_marker', {
+      name: name,
+      icon: icon,
+      pos_x: x,
+      pos_y: y,
+    });
+  };
+
+  const cancelNewMarker = () => {
+    toggleNewMarkerMenu();
+    act('cancel_new_marker');
+  };
+
+  const setImage = (value) => {
+    data.icon = value;
+    data.image = placable_marker_images[value];
+    act('update_icon', { icon: value });
+  };
+
+  const setX = (value) => {
+    data.x = value;
+  };
+
+  const setY = (value) => {
+    data.y = value;
+  };
+
+  return (
+    <Window
+      title={title}
+      theme={theme}
+      width={750}
+      height={390}
+    >
+      <Window.Content>
+        <Stack justify="center">
+          <Stack.Item>
+            <Section
+              title="Minimap"
+              fill
+              buttons={(
+                <Button
+                  icon="undo"
+                  color="green"
+                  content="Reset Map Scale"
+                  onClick={() => act('reset_scale')}
+                />
+              )}>
+              <Flex>
+                <Flex.Item>
+                  <ByondUi
+                    params={{
+                      id: minimap_id,
+                      type: 'map',
+                    }}
+                    style={{
+                      width: "300px",
+                      height: "300px",
+                    }}
+                  />
+                </Flex.Item>
+              </Flex>
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow>
+            <Section
+              title="Minimap Markers"
+              fill
+              buttons={(
+                <Box>
+                  <Button
+                    icon="plus"
+                    color="green"
+                    content="New"
+                    onClick={() => toggleNewMarkerMenu()}
+                  />
+                  <Button
+                    icon={markers_visible ? "eye-slash" : "eye"}
+                    color={markers_visible ? "red" : "green"}
+                    content={markers_visible ? "Hide All" : "Show All"}
+                    onClick={() => act('toggle_visibility_all')}
+                  />
+                </Box>
+              )}>
+              {(!!showNewMarkerMenu) && (
+                <Modal
+                  mr={2}>
+                  <Box>
+                    <Flex>
+                      <Flex.Item backgroundColor="black">
+                        <Image
+                          pixelated
+                          height="150px"
+                          width="150px"
+                          src={`data:image/png;base64,${image}`}
+                        />
+                      </Flex.Item>
+                      <Flex.Item ml="10px">
```

Comment:
Not convinced this would work, but when I see `ml` on a `Flex.Item` I can't help but wonder if a `Stack` (and `Stack.Item`s) would solve it easier?

Reply:
`Stack` does indeed work in place of `Flex`, however it leads to disproportionate spacing of elements, which is undesirable, so I opt to use `Flex` with `ml` in this case.